pkg/controllers/provisioningv2/provisioningcluster/template.go

```diff
@@ -510,8 +510,9 @@ func assign(labels map[string]string, key string, value interface{}) error {
 func rkeCluster(cluster *rancherv1.Cluster) *rkev1.RKECluster {
 	return &rkev1.RKECluster{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      cluster.Name,
-			Namespace: cluster.Namespace,
+			Name:        cluster.Name,
+			Namespace:   cluster.Namespace,
+			Annotations: map[string]string{capr.DeleteMissingCustomMachinesAfterAnnotation: cluster.Annotations[capr.DeleteMissingCustomMachinesAfterAnnotation]},
```

Comment:
Is it guaranteed for `cluster.Annotations` to be non-nil here? Same applies to previous places where we access `Annotations`.

Reply:
FWIW you can index a `nil` map, you just can't add to it, so `cluster.Annotations[capr.DeleteMissingCustomMachinesAfterAnnotation]` is defined as `""`.

Reply:
Ah, should have checked before posting.
If we're OK to define `capr.DeleteMissingCustomMachinesAfterAnnotation` annotation as `""` on `rkecluster` even if it not defined on source cluster then I'm fine with it - but leave it to you guys.

Reply:
Talked to @Oats87 and it actually makes sense to define empty annotation for the cases where we would otherwise want to remove it via Apply.